api/crosscontroller/crosscontroller.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package crosscontroller
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/loggo"
+	names "gopkg.in/juju/names.v2"
+
+	"github.com/juju/juju/api/base"
+	apiwatcher "github.com/juju/juju/api/watcher"
+	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/core/crossmodel"
+	"github.com/juju/juju/watcher"
+)
+
+var logger = loggo.GetLogger("juju.api.crosscontroller")
+
+// Client provides access to the CrossController API facade.
+type Client struct {
+	base.ClientFacade
+	facade base.FacadeCaller
+}
+
+// NewClient creates a new client-side CrossModelRelations facade.
+func NewClient(caller base.APICallCloser) *Client {
+	frontend, backend := base.NewClientFacade(caller, "CrossController")
+	return &Client{
+		ClientFacade: frontend,
+		facade:       backend,
+	}
+}
+
+func (c *Client) ControllerInfo(controllerUUID string) (*crossmodel.ControllerInfo, error) {
+	if !names.IsValidController(controllerUUID) {
+		return nil, errors.NotValidf("controller UUID %q", controllerUUID)
+	}
+	var results params.ControllerAPIInfoResults
+	controllerTag := names.NewControllerTag(controllerUUID)
+	args := params.Entities{
+		Entities: []params.Entity{{
+			controllerTag.String(),
+		}},
+	}
+	if err := c.facade.FacadeCall("ControllerInfo", args, &results); err != nil {
+		return nil, errors.Trace(err)
+	}
+	if len(results.Results) != 1 {
+		return nil, errors.Errorf("expected 1 result, got %d", len(results.Results))
+	}
+	if err := results.Results[0].Error; err != nil {
+		return nil, errors.Trace(err)
+	}
+	info := results.Results[0].Info
+	return &crossmodel.ControllerInfo{
```

Comment:
we now have a controller alias - let's populate that so we can use it when logging errors etc

Reply:
done